lldb/source/Symbol/SwiftASTContext.cpp

```diff
@@ -1464,22 +1464,27 @@ void SwiftASTContext::AddExtraClangArgs(std::vector<std::string> ExtraArgs) {
         clang_argument == "-working-directory")
       continue;
 
+    auto clear_arg = llvm::make_scope_exit([&] { clang_argument.clear(); });
+
     // Enable uniquing for -D and -U options.
     bool is_macro = (clang_argument.size() >= 2 && clang_argument[0] == '-' &&
                      (clang_argument[1] == 'D' || clang_argument[1] == 'U'));
     bool unique = is_macro;
 
     // Consume any -working-directory arguments.
     StringRef cwd(clang_argument);
-    if (cwd.consume_front("-working-directory"))
+    if (cwd.consume_front("-working-directory")) {
       cur_working_dir = cwd;
-    else {
-      // Otherwise add the argument to the list.
-      if (!is_macro)
-        ApplyWorkingDir(clang_argument, cur_working_dir);
-      AddClangArgument(clang_argument.str(), unique);
+      continue;
+    }
+    // Drop -Werror; it would only cause trouble in the debugger.
+    if (clang_argument.startswith("-Werror")) {
+      continue;
     }
-    clang_argument.clear();
+    // Otherwise add the argument to the list.
+    if (!is_macro)
+      ApplyWorkingDir(clang_argument, cur_working_dir);
+    AddClangArgument(clang_argument.str(), unique);
```

Comment:
Instead of using a scope exit to ensure that `clang_argument` is cleared, I expect it'd be simpler (and not much slower) to sink the SmallString<128> declarations into the loop body.

Reply:
The subtle behavior of this loop is to combine multi-arg options, such as `-D MACRO` into one string that is processed as if it were `-DMACRO` all along.

Reply:
Ah, got it. I suppose we could use a flag to indicate when we're processing a multi-arg macro, and then use that flag to avoid the extra ApplyWorkingDir call, but that's about as complicated as the current version.